third-party/sysdeps/linux/grpc/CMakeLists.txt

```diff
@@ -0,0 +1,147 @@
+if(NOT CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR)
+  # When included in TheRock, we download sources and set up the sub-project.
+  set(_source_dir "${CMAKE_CURRENT_BINARY_DIR}/source")
+  set(_download_stamp "${_source_dir}/download.stamp")
+
+  therock_subproject_fetch(therock-grpc-sources
+    SOURCE_DIR "${_source_dir}"
+    GIT_REPOSITORY "https://github.com/grpc/grpc.git"
+    GIT_TAG "v1.76.0"
+    GIT_SHALLOW ON
+    GIT_SUBMODULES_RECURSE ON
+    TOUCH "${_download_stamp}"
```

Comment:
This is a big red flag. Third party deps need to be downloaded as tarballs which we later move to our own S3 bucket. External git repos are a strict no-go, especially if unpinned. This is a security risk and furthermore prevents hermetic builds or rather reproducible builds.

Reply:
@WBobby , we can clone with submodules, upload the result to S3 bucket, and then base grpc build on that.
```
git clone -b v1.76.0 https://github.com/grpc/grpc --depth=1 --shallow-submodules --recurse-submodules grpc
tar czf grpc-1.76.0-source.tar.gz --exclude='grpc/.git' grpc
```